corehq/apps/users/management/commands/add_data_dict_permissions.py

```diff
@@ -0,0 +1,73 @@
+from django.core.management.base import BaseCommand
+
+from django.db.models import Q
+
+from dimagi.utils.chunked import chunked
+from corehq.apps.users.permissions import EXPORT_PERMISSIONS
+from corehq.apps.users.models_role import Permission, UserRole
+from corehq.toggles import DATA_DICTIONARY, DATA_FILE_DOWNLOAD
+
+
+class Command(BaseCommand):
+    help = "Adds data dictionary permission to user role if not already present and edit tab is viewable."
+
+    def build_role_can_edit_commcare_data_q_object(self):
+        edit_data_permission, created = Permission.objects.get_or_create(value='edit_data')
+        return Q(rolepermission__permission_fk_id=edit_data_permission.id)
```

Comment:
Oh yeah, speaking of which, I meant to comment that I don't see anything that explicitly creates the new permission - I see a couple migrations like this:
https://github.com/dimagi/commcare-hq/blob/570ca1849bf833ef39cb4bda2928b5a8a7c44234/corehq/apps/users/migrations/0045_add_view_tableau_permission.py#L6-L7
Is that standard practice whenever a permission is added, or should we expect to create it lazily?

Reply:
> For this permission I agree but it is possible for new permissions (or on systems with no roles but that's irrelevant here) not to exist since they get created lazily.

@snopoke, So is it because we expect the "edit_data" to have been created lazily that we shouldn't use `.get_or_create`? When is it that a permission is actually lazily created if not in the migration?

I am trying to figure out if I should use `.get_or_create` for the other permissions in the file.

Reply:
Ideally we should use ` Permission.create_all()` when creating a new permission which reduces the unknowns in the system.

I do think it's marginally safer not to use 'get_or_create' since that would protect us from typo's.

In terms of how the lazy creation works, it is part of the `ForeignValue` which is used on [RolePermission](https://github.com/dimagi/commcare-hq/blob/5ca8b07ee01eec052c2f10f3789a93e1549a1dd4/corehq/apps/users/models_role.py#L260). The create happens here: https://github.com/dimagi/commcare-hq/blob/61cf908965820ed5b4d13dfa9d9ace7274a483dc/corehq/util/models.py#L338

For docs on ForeignValue see https://github.com/dimagi/commcare-hq/blob/61cf908965820ed5b4d13dfa9d9ace7274a483dc/corehq/util/models.py#L264-L279

Reply:
used`Permission.create_all()' and replaced `.get_or_create` in commit: c80d2b8fd3eb327db2ed493d80eb50b7c5669247
and refactored it out in 17a4ea948cf60f0cec33cbdfa34b694850a48c4e